### ISSUE TITLE
Assetion in `std::string buffer::get_line(bool skip_whitespace)`, with MSVC build

### ISSUE THREAD

**Issue by Try** [lmichaelis/phoenix#36]
![изображение](https://user-images.githubusercontent.com/122894/201543875-016e82a6-9372-4a8a-a85a-46da6faafc8d.png)
sorry for awful screenshot :)

Offended code:
```
	std::string buffer::get_line(bool skip_whitespace) {
		...
		if (skip_whitespace) {
			auto count = mismatch([](char chr) { return !std::isspace(chr); }); // <--
			if (count == -1) {
				position(limit()); // the end of the buffer has been reached
			} else {
				position(position() + count);
			}
		}

		return tmp;
	}
```
`char` can be signed on some C++ implementations, and have negative values.
Issue was hit, when parsing world mesh in steam-eng version of gothic2

**Comment by lmichaelis** [comment]
Hm interesting. To my knowledge `isspace` is supposed to take an `int` anyways (see [cppreference](https://en.cppreference.com/w/cpp/string/byte/isspace))? Regardless I'll see if I can reproduce on MinGW on Linux so that I can actually see what's going on here :)

**Comment by lmichaelis** [comment]
Fixed in 69cd380.
